wm.c

```diff
@@ -1,20 +1,37 @@
 #include <Windows.h>
 #include <signal.h>
 #include <stdlib.h>
+#include <stdio.h>
 
 HHOOK hookHandle;
 
 void ctrlc(int sig) {
-    UnhookWindowsHookEx(hookHandle);
+    if (!UnhookWindowsHookEx(hookHandle)) {
+        wprintf(L"[!] Error while unhooking hookHandle.\nError: %d\n", GetLastError());
+    }
+    
     exit(0);
 }
 
 int main() {
     HMODULE wmDll = LoadLibraryW(L"wm_dll");
+
+    if (wmDll == NULL) {
+        wprintf( L"[!] Error while loading library.\n Error: %ld", GetLastError());
+    }
+
     FARPROC shellProc = GetProcAddress(wmDll, "ShellProc");
 
+    if (shellProc == NULL) {
+        wprintf(L"[!] Error while getting process address.\nError: %ld", GetLastError());
```

Comment:
Here I would recommend using something like goto cleanup since we want to free the DLL and return with an error both here and in the next failure check of hookHandle - you can read about the goto cleanup pattern on the internet, its pretty much the only good use for goto in c in my opinion :)